src/server/mod.rs

```diff
@@ -124,11 +124,11 @@ impl LspServerState {
     /// Get a composition from the state
     ///
     /// We return a copy here, as the pointer across threads isn't supported.
-    pub fn get_composition(
-        &self,
+    pub fn get_mut_composition(
+        &mut self,
         uri: &lsp::Url,
-    ) -> Option<composition::Composition> {
-        self.compositions.get(uri).cloned()
+    ) -> Option<&mut composition::Composition> {
+        self.compositions.get_mut(uri)
     }
```

Comment:
1. Why you decide to use mut instead of returning a copy?
2. Should the comment get updated as well since the return is no longer a copy but a mutable reference?

Reply:
The composition in behind a mutex, as the shared object. The existing code was:
1. acquiring the lock
2. cloning a copy.
3. releasing the lock...at which point anything else could overwrite the object.
4. updating the copy.
5. reacquire the lock --> then set in state.

The was the mutex should instead be used, is to acquire the lock on the mutable object, hold the object as short as possible (changing the object), then release the lock. That was the main point of the change made in [this commit](https://github.com/influxdata/flux-lsp/pull/616/commits/97066c8997ef377fe99006ac00a06430d48d2ac6).